internal/service/resourcepolicy/model.go

```diff
@@ -52,7 +53,8 @@ func NewUserMetadataObjectType(ctx context.Context, input *admin.ApiAtlasUserMet
 }
 
 func NewPolicyObjectType(ctx context.Context, input *[]admin.ApiAtlasPolicy) (*types.List, diag.Diagnostics) {
-	if input == nil {
+	var nilPointer *[]admin.ApiAtlasPolicy
```

Comment:
Thank you for spotting this @lantoli

Reply:
@EspenAlbert i was thinking more about this and I don't think this applies here, as the problem is only when using interfaces, and that's not the case here.  
I think nilPointer and nil will be the same in this case